components/automate-ui/src/app/modules/desktop/desktop-detail/desktop-detail.component.ts

```diff
@@ -1,42 +1,85 @@
-import { Component, Input, Output, EventEmitter } from '@angular/core';
-import { Desktop } from 'app/entities/desktop/desktop.model';
+import { Component, OnInit, OnDestroy, Input, Output, EventEmitter } from '@angular/core';
+import { Desktop, DailyNodeRunsStatus } from 'app/entities/desktop/desktop.model';
+import { takeUntil } from 'rxjs/operators';
 import { DateTime } from 'app/helpers/datetime/datetime';
-
-const historyTypes = [
-  { text: 'converged', icon: 'check_box' },
-  { text: 'unchanged', icon: 'indeterminate_check_box' },
-  { text: 'error', icon: 'error' },
-  { text: 'unknown', icon: 'help' }
-];
-
-const checkinHistory = Array.from(new Array(29)).map((_v, i) => {
-  const historyType = historyTypes[Math.floor(Math.random() * historyTypes.length)];
-  const date = new Date();
-  return { ...historyType, date: new Date(date.setDate(date.getDate() - i)) };
-});
+import { Store } from '@ngrx/store';
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { Subject, Subscription } from 'rxjs';
+import { GetDailyNodeRunsStatusTimeSeries } from 'app/entities/desktop/desktop.actions';
+import { getDailyNodeRuns } from 'app/entities/desktop/desktop.selectors';
 
 @Component({
   selector: 'app-desktop-detail',
   templateUrl: './desktop-detail.component.html',
   styleUrls: ['./desktop-detail.component.scss']
 })
-export class DesktopDetailComponent {
+export class DesktopDetailComponent implements OnInit, OnDestroy {
 
   @Input() desktop: Desktop;
   @Input() fullscreened = false;
 
   @Output() closed: EventEmitter<any> = new EventEmitter();
   @Output() fullscreenToggled: EventEmitter<void> = new EventEmitter();
 
+  public checkInHistory: DailyNodeRunsStatus[];
   public ceil = Math.ceil;
   public DateTime = DateTime;
   public showCheckinDebug = false;
   public checkinTableType = 'grid';
   public checkinGridFlexType = 'wrap';
   public checkinNumDays = 14;
+  // These are Material Icon names from https://material.io/resources/icons/
+  public historyIcons = {
+    converged: 'check_box',
+    unchanged: 'indeterminate_check_box',
+    failure: 'warning',
+    error: 'warning',
+    unknown: 'help',
```

Comment:
Is it possible to use the Carbon icon for "unknown-filed" here? https://www.carbondesignsystem.com/guidelines/icons/library/ (you can search "unknown" there)

Reply:
@apriofrost we don't have carbon icons available in the UI yet. That particular package has been problematic 😞

Reply:
@apriofrost For right now we do not have carbon icons in our repo, we will revisit.

Reply:
@apriofrost i shared this in the ux team a few days ago, we need to give the ui team a folder of svgs that we want to use so they dont have to deal with the carbon icon package. its on my list of things to do.

Reply:
cool, we can settle on this one for now. Thank you all!